Create a getting started with scpca data section (#59)
AlexsLemonade/scpca-docs: docs/getting_started.md

```

Here we are calculating PCA by using the default of the top 500 most highly variable genes as input, however this is not always the optimal choice. 
We encourage you to visit the [Feature selection chapter in Orchestrating Single Cell Analysis](http://bioconductor.org/books/3.13/OSCA.basic/feature-selection.html#feature-selection) to read more about modeling gene variance and selecting the highly variable genes. 

PCA is commonly used for initial dimensionality reduction. 
However, we can use more advanced techniques, like [UMAP (Uniform Manifold Approximation and Projection)](http://bioconductor.org/books/3.13/OSCA.basic/dimensionality-reduction.html#uniform-manifold-approximation-and-projection), that may be better for visualization. 
Use caution when interpreting UMAP results, as location, spacing, and density of clusters can be dependent on parameter choices and random effects and do not always accurately represent the relationships among cells.

UMAP can also be quite slow for a large dataset, so we can use the previous PCA results as input to speed up the analysis. 

```r
# Run UMAP using already stored PCA results

> normalized_sce <- scater::runUMAP(normalized_sce, 
>                           dimred = "PCA")

Reviewer: Spacing was off, but I think we can do one line here anyway
```suggestion
normalized_sce <- scater::runUMAP(normalized_sce, dimred = "PCA")
```

There are two RDS files that will be available for each library, a `filtered.rds` and an `unfiltered.rds` file. 
The `filtered.rds` files will contain the gene expression data for only droplets that are likely to contain cells, removing any potential empty droplets. 
The `unfiltered.rds` file will contain the gene expression data for all droplets, regardless of the presence of a cell or not. 
See the {ref}`section on filtering cells<processing_information:filtering cells>` for more information on how we remove potential empty droplets. 
In most scenarios, we recommend starting with the `filtered.rds` file. 

The first step in analyzing the provided gene expression data will be to import the data into R. 
As a reminder, each RDS file contains a [`SingleCellExperiment` object](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html). 
Refer to {ref}`single-cell gene expression file contents<sce_file_contents:single-cell gene expression file contents>` for a more detailed description on the contents of the included `SingleCellExperiment` object. 

> In order to work with `SingleCellExperiment` objects in R, we need to ensure that we have the [`SingleCellExperiment` package]() installed and loaded. 

Reviewer 2: This line needs a URL. I assume you intended to include this one.

```suggestion
In order to work with `SingleCellExperiment` objects in R, we need to ensure that we have the [`SingleCellExperiment` package](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html) installed and loaded. 
```

The below commands can be used to import the RDS file into R and save the `SingleCellExperiment` object. 

```r
# if SingleCellExperiment is not installed, install the package
# otherwise the installation step can be skipped

> BiocManager::install("SingleCellExperiment")

Reviewer 2: You could consider putting this inside some logic. 

```suggestion
if (!("SingleCellExperiment" %in% installed.packages())) {
  BiocManager::install("SingleCellExperiment")
}
```

library(SingleCellExperiment)
# read in the RDS file, including the complete path to the file's location

> sce <- readRDS("SCPCL000000_filtered.rds")

Reviewer 2: I wonder if you need to put more information about the path outside of the comment and mention that you're using a placeholder. The way you have the example here assumes that they moved the file outside of the folder structure in the download, right? That seems a little odd to me. I know it's slightly different between project and sample downloads, but there will always be a `sample_id` folder IIRC.

```

More resources for learning about `SingleCellExperiment` objects: 

- [`SingleCellExperiment` objects Vignette](https://bioconductor.org/packages/devel/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html) on working with `SingleCellExperiment` objects
- [Orchestrating Single Cell Analysis chapter on the `SingleCellExperiment` class](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html)

## Quality control  

> After we have imported the RDS file into R and accessed the `SingleCellExperiment` object, we can begin working with the data. 

Reviewer 2: What does "accessed the `SingleCellExperiment` object" mean?

> Before we perform any downstream steps, it is recommended to remove any low quality cells from our dataset.

Reviewer 2: ```suggestion
Before we perform any downstream steps, we recommend removing low quality cells from the dataset.
```

> This would include cells that may be dying or damaged, showing a higher percentage of reads coming from mitochondrial cells. 
> Low quality cells may also be present due to technical artifacts such as inefficient reverse transcription or PCR amplification resulting in a lower number of total reads and unique genes identified. 

Reviewer 2: Don't need to use this exact wording, but it's helpful to start with _what you're filtering out_ and then say _why_ in my opinion.

```suggestion
Low quality cells include those with a higher percentage of reads from mitochondrial genes (i.e., those that are damaged or dying) and those with a lower number of total reads and unique genes identified (i.e., those with inefficient reverse transcription or PCR amplification). 
```

> For the filtered `SingleCellExperiment` object present in all `filtered.rds` files, we have used [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a data driven approach to predict low-quality cells, to identify cells that should be removed prior to downstream analyses. 

Reviewer 2: ```suggestion
All `filtered.rds` objects include [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html) results that can be used to identify cells that should be removed prior to downstream analyses. 
```

> `miQC` jointly models the proportion of mitochondrial reads and the number of unique genes detected in each cell to calculate the probability of a cell being compromised (i.e. dead or damaged). 

Reviewer 2: ```suggestion
`miQC` jointly models the proportion of mitochondrial reads and the number of unique genes detected in each cell to calculate the probability of a cell being compromised (i.e., dead or damaged). 
```

High-quality cells are those with a low probability of being being compromised (< 0.75) or sufficiently low mitochondrial content. 

> We have already incorporated the `miQC` results into the {ref}`colData() of the SingleCellExperiment object<sce_file_contents:cell metrics>`. 

Reviewer 2: I would maybe move this link to the `colData()` section to the sentence in my first suggestion.

All cells that are identified as low-quality cells will have `FALSE` in the `miQC_pass` column of the `colData()` and can be removed prior to downstream analyses. 

> The following command can be used to remove those cells: 

Reviewer: A small clarification
```suggestion
The following command can be used to the low-quality cells: 
```

```r
# create a vector of barcode names for cells that are not compromised 
# the `$` notation denotes access to the colData slot of the `SingleCellExperiment` object 
cells_to_keep <- colnames(sce$miQC_pass == TRUE)

# filter the `SingleCellExperiment`
filtered_sce <- sce[, cells_to_keep]
```

> For more information on using `miQC` for filtering cells, see the following resources: 

Reviewer 2: Can you put a line break between these two lines please? For readability 👀

- [`miQC` vignette](https://bioconductor.org/packages/release/bioc/vignettes/miQC/inst/doc/miQC.html)
- [Hippen _et al._ 2021](https://doi.org/10.1371/journal.pcbi.1009290)

Alternatively, you can directly filter cells based on having a minimum number of unique genes, total reads, and a maximum fraction of mitochondrial reads. 
Using the function, [`scuttle::addPerCellQCMetrics()`](https://rdrr.io/github/LTLA/scuttle/man/addPerCellQCMetrics.html), we can calculate the total RNA counts, unique genes, and mitochondrial fraction for each cell and store the results in the `colData()`. 
These metrics can be used to directly filter the `SingleCellExperiment` object based on informed minimum thresholds. 

> The `filtered.rds` files contain `SingleCellExperiment` objects with these metrics added to the `colData()` and you can read more about all metrics that are included in the {ref}`cell metrics section of the SingleCellExperiment file contents<sce_file_contents:cell metrics>`.

Reviewer 2: So these results are already in the object right? (I am under the impression that you need to add the QC metrics _before_ you run `miQC`.) Start with the fact that these are already present and you can use them to filter.

If you are planning to filter low quality cells using these hard cutoffs, we encourage you to read more about the various metrics and plot the distribution of each metric before deciding on which cells to exclude. 
The chapter on [Quality Control in Orchestrating Single Cell Analysis](http://bioconductor.org/books/3.13/OSCA.basic/quality-control.html#quality-control) provides a nice guide to checking diagnostic plots and then choosing cutoffs. 

## Normalization 

Single-cell RNA sequencing inherently comes with technical confounders that, if not accounted for, can impact the interpretation of downstream analysis. 
Technical variables include differences in sequencing depth and cell size across cells in a single-cell library ([Stegle _et al._ 2015](https://doi.org/10.1038/nrg3833).)

If we were to proceed with downstream analysis such as differential gene expression and cell clustering without accounting for such effects, these technical problems could impact the overall conclusions.

> We need to normalize our data to account for the technical variation between cells in our data and maximize biological variation, allowing us to make more accurate conclusions from downstream analysis.

Reviewer 2: I'm torn about whether we need this much information here or if it's better to get right into what we recommend. We do provide links out to other resources.

Reviewer 2: > I tend to think this document should have less rather than more

I agree with this take and would extend it to maybe mean cutting the downstream analysis portion.

Author: To clarify, do you mean completely remove the downstream analysis portion, or to just simplify it?

Reviewer 2: I meant completely remove, but after taking a look at @jashapiro's comments I propose: you start by responding to his comments (e.g., removing feature selection) and then we'll see if we want to cut further.

> These metrics can be used to directly filter the `SingleCellExperiment` object based on informed minimum thresholds. 
> 
> If you are planning to filter low quality cells using these hard cutoffs, we encourage you to read more about the various metrics and plot the distribution of each metric before deciding on which cells to exclude. 

Reviewer: Seemed like this could all be one paragraph. I also removed `minimum` since for at least one people are likely to use a max. 

```suggestion
These metrics can be used to directly filter the `SingleCellExperiment` object based on informed thresholds. 
If you are planning to filter low quality cells using such thresholds, we encourage you to read more about the various metrics and plot the distribution of each metric before deciding on which cells to exclude. 
```
